.editorconfig

```diff
@@ -1,20 +1,25 @@
+# Top most .editorconfig file
 root = true
 
+# Unix-style newlines with a newline ending every file
 [*]
+charset = utf-8
 end_of_line = lf
+indent_size = 4
+indent_style = space
 insert_final_newline = true
 trim_trailing_whitespace = true
-charset = utf-8
-indent_style = space
-indent_size = 4
 
-[*{.cc,h}]
+# Tab indentation for shell scripts, Dockerfiles and Makefiles
+[{*.{bash,sh},Dockerfile,Dockerfile.*,Makefile,*.mk}]
 indent_style = tab
-indent_size = 4
+keep_padding = true # Unsupported in some editors
```

Comment:
I believe comments after declarations are non-standard (or at least my editor thinks so)

So it's probably best to put comments on it's own line